sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -232,10 +232,12 @@ public void run() {
             } while (spansList.isEmpty());
           }
           spansCopy = new ArrayList<>(spansList);
-          spansList.clear();
         }
         // Execute the batch export outside the synchronized to not block all producers.
-        exportBatches(spansCopy);
+        int taken = onBatchExport(spansCopy);
```

Comment:
I might be missing it, but I think this will cause multiple batches to be sent concurrently? If so, I think it's not respecting the spirit of `maxExportBatchSize`. To be honest, I don't know exactly what the spirit of `maxExportBatchSize` is but I think it's one of 1) send batches sequentially or 2) wait `scheduledDelayMillis` between each chunk of `maxExportBatchSize` no matter what. #1565 aims for 1), neither of our PRs aim for 2) :( But I guess if we're sending concurrently that's not good for BSP, that's the SSP's role. Let me know if I'm confused though.

Reply:
It isn’t sending an export concurrently. Only one export is permitted at a time within the onExportBatch function.

Reply:
Exporting “no matter what” on schedule is what things look like before this PR. ;-)

Reply:
Ah I see how the return 0 works. Isn't this effectively a spinlock because we'll keep on looping, find the queue is larger than batch size, try to export, find it's not available, and continue until it's available again? I wouldn't expect to see spinning to wait for I/O, is it better than parking?

Reply:
Good point. It would indeed spin until the number of buffer elements drops below the max buffer size. I think we should move the condition on the wait also... that way it wouldn't spin. The batch would then just wake up on its timer no matter what and then try another export. This would ultimately represent the park that you suggested.

I will push another commit now along with some other small changes to improve comprehension.